### ISSUE TITLE
Not all members in Multicastclient are disposed

### ISSUE THREAD

**Issue by fgheysels** [richardschneider/net-mdns#93]
When instantiating a `MulticastService` object, I see that 2 `UdpClient` instances are created.   However, those 2 instances are not disposed when disposing the `MulticastService` object.   Since `UdpClient` implements `IDisposable`, I believe those 2 instances should be disposed as well ?

It seems that not disposing the UdpClient instances causes a memory leak.

**Comment by fgheysels** [comment]
I've created pull-request [94 ](https://github.com/richardschneider/net-mdns/pull/94)for this.

**Comment by fgheysels** [comment]
@richardschneider : Do you have any chance to take a look at this issue and the corresponding [PR](https://github.com/richardschneider/net-mdns/pull/94) since this is a bug that is hurting us. 

Thanks in advance

**Comment by queequac** [comment]
No contributions from @richardschneider for months now, no comments on issues. :(
Temporarily I am using a local built with this and my fix, since I don't want to create a competing fork.

**Comment by fgheysels** [comment]
We're using a local build as wel ftm, but I'd like to get rid of it sooner then later :)

Hope @richardschneider is doing well.

**Comment by fgheysels** [comment]
> No contributions from @richardschneider for months now, no comments on issues. :(
> Temporarily I am using a local built with this and my fix, since I don't want to create a competing fork.

Maybe we should, since it's almost 6 months since @richardschneider was active on github

**Comment by queequac** [comment]
@fgheysels In the meanwhile I also think at least a temporary fork might indeed be a good idea. Are you familiar with GitHub Actions and publishing NuGet packages?

**Comment by fgheysels** [comment]
No, I'm not familiar with GitHub Actions, I mostly use Azure DevOps which also integrates well with Github repos.

**Comment by queequac** [comment]
The question remains open, how we would actually do this.
Who owns the fork and creates the NuGet packages? How to name those packages to make clear its a fork and not the official ones from @richardschneider, etc.?

**Comment by fgheysels** [comment]
Should we fork it to a separate GitHub account instead of to one of our personal accounts ?
Do you have any experience with GitHub actions ?  If not, I see 2 options: or we use Azure DevOps, or we learn something new and investigate on GitHub actions :)

**Comment by queequac** [comment]
I have also neither worked with GitHub Actions, nor published packages to NuGet. So it would more be the "learn something new" approach. ;)

Aside from this, I'd also prefer a separate account/organization rather than using our personal accounts. Nevertheless I wrote an email to Richard some minutes ago, just to give him yet another chance to stop us. And maybe we can switch to some other tool or good old email for further disucssions instead of flooding this issue.
